build.gradle.kts

```diff
@@ -71,107 +73,208 @@ tasks.jar.configure {
-  dependsOn(tasks.javadoc)
-  from(tasks.javadoc.map { it.destinationDir!! })
-  archiveClassifier.set("javadoc")
+tasks.withType<PublishToMavenRepository>().configureEach {
+  dependsOn(cleanTempRepo)
 }
 
-project.afterEvaluate {
-  tasks.register("writePom") {
-    val outputFile = file("$buildDir/pom/${project.name}-${project.version}.pom")
-    outputs.file(outputFile)
+// ./gradlew publishToTmpMaven would publish the files to build/repo, so they can be signed and released
+val publishToTmpMaven by tasks.registering {
+  description = "Publish Maven publications to $tempRepoPath"
+  group = PublishingPlugin.PUBLISH_TASK_GROUP
+  dependsOn("publishAllPublicationsToTmpMavenRepository")
+}
 
-    doLast {
-      maven {
+publishing {
+  repositories {
+    maven {
+      name = "tmpMaven"
+      url = uri(tempRepoPath)
+    }
+  }
+  publications {
+    // java-gradle-plugin adds publication late, so we need afterEvaluate here
+    afterEvaluate {
+      // Configure <name> and <description> for the java module of the plugin
+      named<MavenPublication>("pluginMaven") {
         pom {
-          project {
-            withGroovyBuilder {
-              "name"("App Engine Gradle Plugin")
-              "description"("This Gradle plugin provides tasks to build and deploy Google App Engine applications.")
-
-              "url"("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
-              "inceptionYear"("2016")
-
-              "scm" {
-                "url"("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
-                "connection"("scm:https://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
-                "developerConnection"("scm:git://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
-              }
-
-              "licenses" {
-                "license" {
-                  "name"("The Apache Software License, Version 2.0")
-                  "url"("http://www.apache.org/licenses/LICENSE-2.0.txt")
-                  "distribution"("repo")
-                }
-              }
-              "developers" {
-                "developer" {
-                  "id"("loosebazooka")
-                  "name"("Appu Goundan")
-                  "email"("appu@google.com")
-                }
-              }
-            }
+          name.set("App Engine Gradle Plugin")
+          description.set("This Gradle plugin provides tasks to build and deploy Google App Engine applications.")
+        }
+      }
+    }
+    withType<MavenPublication> {
+      // Use the resolved versions in pom.xml
+      // Gradle might have different resolution rules, so we set the versions
+      // that were used in Gradle build/test.
+      versionMapping {
+        usage(Usage.JAVA_RUNTIME) {
+          fromResolutionResult()
+        }
+        usage(Usage.JAVA_API) {
+          fromResolutionOf("runtimeClasspath")
+        }
+      }
+      pom {
+        // <editor-fold defaultstate="collapsed" desc="Default POM values">
```

Comment:
These regions are awesome, they're much better than the raw comments, or even `region`/`endregion` pair for IDEA. Where does the support for this in IDEA come from?